gh-131146: Fix month names in a genitive case in calendar module (#131147)
python/cpython: Doc/library/calendar.rst

.. data:: month_name

   A sequence that represents the months of the year in the current locale.
   This follows normal convention of January being month number 1, so it has

>    a length of 13 and ``month_name[0]`` is the empty string.

Reviewer: This is just a formatting change and should not be included in this PR.

Author: It's to limit the line length to 80 characters in accordance with the [reStructuredText markup guideline](https://devguide.python.org/documentation/markup/)

Reviewer: Again, you did not modify this section, please revert it. Existing violations do not have to be fixed, I assure you there are hundreds in the file.

Author: Reverted

       >>> import calendar
       >>> list(calendar.month_name)
       ['', 'January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']

   .. caution::

>       In locales with alternative forms of month names, the :data:`!month_name` sequence

Reviewer: ```suggestion
      In locales with alternative month names forms, the :data:`!month_name` sequence
```
Simpler rewording

Author: Fixed, thanks.

      may not be suitable when a month name stands by itself and not as part of a date.
      For instance, in Greek and in many Slavic and Baltic languages, :data:`!month_name`
      will produce the month in genitive case. Use :data:`standalone_month_name` for a form
      suitable for standalone use.

.. data:: month_abbr

   A sequence that represents the abbreviated months of the year in the current
   locale. This follows normal convention of January being month number 1, so

>    it has a length of 13 and  ``month_abbr[0]`` is the empty string.

Reviewer: Again unrelated formatting changes

Author: It's to limit the line length to 80 characters in accordance with the [reStructuredText markup guideline](https://devguide.python.org/documentation/markup/).

Reviewer: You did not modify this section, it is unrelated to this pr, please revert it.

Author: Reverted

       >>> import calendar
       >>> list(calendar.month_abbr)
       ['', 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']

   .. caution::

      In locales with alternative forms of month names, the :data:`!month_abbr` sequence
      may not be suitable when a month name stands by itself and not as part of a date.
      Use :data:`standalone_month_abbr` for a form suitable for standalone use.

.. data:: standalone_month_name

   A sequence that represents the months of the year in the current locale
   in the grammatical form used when a month name stands by itself if the locale

>    provides one. If the locale does not supply an alternative form, it is equal to

Reviewer: ```suggestion
   provides one. If the locale does not supply a standalone form, it is equal to
```
Maybe this would be clearer?

Author: Thanks, fixed

>    provides one. If the locale does not supply an alternative form, it is equal to

Reviewer: ```suggestion
   A sequence that represents the months of the year in the current locale
   in the standalone form if the locale provides one. Else it is equivalent to
```
Much simpler format, as docs should be [e.g.](https://developer.apple.com/documentation/foundation/date/formatstyle/symbol/standalonemonth)

Author: Updated

>    A sequence that represents the months of the year in the current locale
>    in the grammatical form used when a month name stands by itself if the locale
>    provides one. If the locale does not supply an alternative form, it is equal to

Reviewer: ```suggestion
   A sequence that represents the months of the year in the current locale
   in the standalone form if the locale provides one. Else it is equivalent to
```
Much simpler format, as docs should be [e.g.](https://developer.apple.com/documentation/foundation/date/formatstyle/symbol/standalonemonth)

Author: Updated

   :data:`month_name`.

       >>> import calendar
       >>> list(calendar.standalone_month_name)
       ['', 'January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']

> 

Reviewer: This example is pointless in the English locale, and it links to `month_name` with the exact same example.

Author: I agree, it's not relevant for English. But which locale should I use here?

Reviewer: We do not have to provide an example.

Author: Removed

   .. versionadded:: next

.. data:: standalone_month_abbr

   A sequence that represents the abbreviated months of the year in the current
   locale in the grammatical form used when a month name stands by itself if the
   locale provides one. If the locale does not supply an alternative form, it is equal to
   :data:`month_abbr`.

       >>> import calendar
       >>> list(calendar.standalone_month_abbr)
       ['', 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']

> 

Reviewer: Ditto

Author: Removed

.. data:: month_name

   A sequence that represents the months of the year in the current locale

>    in the grammatical form used when the month is part of a complete date.

Reviewer 2: It might be better to move these notes about grammar to a new paragraph or admonition. I fear that they are more confusing than helpful to a reader of the English docs using Python in English, as the language has no differentiation. In the same vein, the note should make it clear that this is for localised use of the module.

Author: @AA-Turner Will a caution like this work?

    In locales with alternative forms of month names, the :data:`month_name` sequence
    may not be suitable when a month is named by itself and not as part of a date. For instance,
    in Greek and in many Slavic and Baltic languages, :data:`month_name` will produce
    the month in genitive case. Use :data:`alt_month_name` for a form suitable for standalone use.

Reviewer 3: Maybe `when a month name stands by itself` instead of `when a month is named by itself`? Otherwise it sounds good.

.. data:: month_abbr

   A sequence that represents the abbreviated months of the year in the current
   locale in the grammatical form used when the month is part of a complete date.
   This follows normal convention of January being month number 1, so it has
   a length of 13 and  ``month_abbr[0]`` is the empty string.

       >>> import calendar
       >>> list(calendar.month_abbr)
       ['', 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']

> 
> .. data:: alt_month_name
> 
>    A sequence that represents the months of the year in the current locale
>    in the grammatical form used when the month is named by itself if the locale
>    provides one. If the locale does not supply an alternative form, it falls back
>    to the behavior of :data:`month_name`.
> 
>        >>> import calendar
>        >>> list(calendar.alt_month_name)
>        ['', 'January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']
> 
> 
> .. data:: alt_month_abbr
> 
>    A sequence that represents the abbreviated months of the year in the current
>    locale in the grammatical form used when the month is named by itself if the
>    locale provides one. If the locale does not supply an alternative form, it falls
>    back to the behavior of :data:`month_abbr`.
> 
>        >>> import calendar
>        >>> list(calendar.alt_month_abbr)
>        ['', 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
> 
> 

Reviewer 4: ```suggestion

.. data:: alt_month_name

   A sequence that represents the months of the year in the current locale
   in the grammatical form used when the month is named by itself if the locale
   provides one. If the locale does not supply an alternative form, it falls back
   to the behavior of :data:`month_name`.

       >>> import calendar
       >>> list(calendar.alt_month_name)
       ['', 'January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']

   .. versionadded:: next

.. data:: alt_month_abbr

   A sequence that represents the abbreviated months of the year in the current
   locale in the grammatical form used when the month is named by itself if the
   locale provides one. If the locale does not supply an alternative form, it falls
   back to the behavior of :data:`month_abbr`.

       >>> import calendar
       >>> list(calendar.alt_month_abbr)
       ['', 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']

   .. versionadded:: next

```

Author: Thanks for pointing out! I've added these tags in the recent commit.

.. data:: month_abbr

   A sequence that represents the abbreviated months of the year in the current
   locale in the grammatical form used when the month is part of a complete date.
   This follows normal convention of January being month number 1, so it has
   a length of 13 and  ``month_abbr[0]`` is the empty string.

       >>> import calendar
       >>> list(calendar.month_abbr)
       ['', 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']

> .. data:: alt_month_name

Reviewer 2: `alt` is not very descriptive (and could be confused for the abbreviated form). Perhaps `nominative_month_name`?

Reviewer 4: I thougzht so as well but GNU specifically uses "alternative" as terminology:

>  The %OB and %Ob formats are a C2X feature, specified in C2X to use the locale’s ‘alternative’ month name; the GNU C Library extends this specification to say that the form used in a complete date is the default and the form naming the month by itself is the alternative. 

See https://www.gnu.org/software/libc/manual/html_node/Formatting-Calendar-Time.html.

Reviewer 5: Not too familiar with all the possibilities here but not all languages where this matters might have a nominative case; e.g. some languages use an ergative-absolutive system instead.

Author: In the `strftime` context, it is usually referred to as an 'alternative form.' It is not always nominative, as many languages do not have a nominative form. The idea is that it a grammatical form used when the month is named by itself. In some languages it could be the nominative form or whatever else.

Author: May be `standalone_month_name` ?

Reviewer 3: I like `standalone_month_name`, it's clearer

   A sequence that represents the months of the year in the current locale
   in the grammatical form used when the month is named by itself if the locale

>    provides one. If the locale does not supply an alternative form, it falls back
>    to the behavior of :data:`month_name`.

Reviewer 3: ```suggestion
   provides one. If the locale does not supply an alternative form, it is equal to :data:`month_name`.
```

       >>> import calendar
       >>> list(calendar.alt_month_name)
       ['', 'January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']

   .. versionadded:: next

.. data:: alt_month_abbr

   A sequence that represents the abbreviated months of the year in the current
   locale in the grammatical form used when the month is named by itself if the

>    locale provides one. If the locale does not supply an alternative form, it falls
>    back to the behavior of :data:`month_abbr`.

Reviewer 3: ```suggestion
   locale provides one. If the locale does not supply an alternative form, it is equal to :data:`month_abbr`.
```
